### PR TITLE
VACCINECER-2137 Fix validation for test product

### PR DESCRIPTION
**Pull request by iris-hunkeler**

### CODE REVIEW

File: src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificateDtoMapperService.java [hunk 1]
```diff
@@ -59,13 +82,13 @@ public VaccinationTouristCertificatePdf toVaccinationTouristCertificatePdf(Vacci
...
     public TestCertificateQrCode toTestCertificateQrCode(TestCertificateCreateDto createDto) {
-        var testValueSet = valueSetsService.getIssuableTestDto(createDto.getTestInfo().get(0));
+        var testValueSet = valueSetsService.validateAndGetIssuableTestDto(createDto.getTestInfo().get(0).getTypeCode(), createDto.getTestInfo().get(0).getManufacturerCode());
         return TestCertificateQrCodeMapper.toTestCertificateQrCode(createDto, testValueSet);
 
     }
 
     public TestCertificatePdf toTestCertificatePdf(TestCertificateCreateDto createDto, TestCertificateQrCode qrCodeData) {
-        var testValueSet = valueSetsService.getIssuableTestDto(createDto.getTestInfo().get(0));
+        var testValueSet = valueSetsService.validateAndGetIssuableTestDto(createDto.getTestInfo().get(0).getTypeCode(), createDto.getTestInfo().get(0).getManufacturerCode());
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificateDtoMapperService.java:91]: Repeatedly using "createDto.getTestInfo().get(0)" should be replaced by extraction to value and referencing, imho

File: src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificatePdfGenerateRequestDtoMapperService.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public VaccinationTouristCertificatePdf toVaccinationTouristCertificatePdf(Vacci
     }
 
     public TestCertificatePdf toTestCertificatePdf(TestCertificatePdfGenerateRequestDto pdfGenerateRequestDto) {
-        var testValueSet = valueSetsService.getIssuableTestDto(pdfGenerateRequestDto.getDecodedCert().getTestInfo().get(0).getTypeOfTest(), pdfGenerateRequestDto.getDecodedCert().getTestInfo().get(0).getTestManufacturer());
+        var testValueSet = valueSetsService.validateAndGetIssuableTestDto(pdfGenerateRequestDto.getDecodedCert().getTestInfo().get(0).getTypeOfTest(), pdfGenerateRequestDto.getDecodedCert().getTestInfo().get(0).getTestManufacturer());
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificatePdfGenerateRequestDtoMapperService.java:68]: Repeatedly using "pdfGenerateRequestDto.getDecodedCert().getTestInfo().get(0)" should be replaced by extraction to value and referencing, imho

**Review by Armin-Isenring-Bit — APPROVED** [review]
The repeated use of .get(0) on a list is an obstacle of a developer's taste. Up to you.

File: src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificateDtoMapperService.java (hunk 1 above)
Comment (iris-hunkeler) [reply src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificateDtoMapperService.java]: I have applied this suggestion and refactored the affected code in this class.

**Review by Armin-Isenring-Bit — APPROVED** [review]
Thanks. Looks good now.

**Pull request merged**